tests/experimental/datasources/conftest.py

```diff
@@ -115,3 +123,139 @@ def inject_engine_lookup_double(monkeypatch: MonkeyPatch) -> ExecutionEngineDoub
     finally:
         for source, engine in original_engine_override.items():
             source.execution_engine_override = engine
+
+
+def pandas_data(
+    context: AbstractDataContext,
+) -> tuple[AbstractDataContext, Datasource, DataAsset, BatchRequest]:
+    csv_path = (
+        pathlib.Path(__file__) / "../../../test_sets/taxi_yellow_tripdata_samples"
+    ).resolve()
+    pandas_ds = context.sources.add_pandas(name="my_pandas")
+    asset = pandas_ds.add_csv_asset(
+        name="csv_asset",
+        base_directory=csv_path,
+        regex=r"yellow_tripdata_sample_(?P<year>\d{4})-(?P<month>\d{2})\.csv",
+        order_by=["year", "month"],
+    )
+    batch_request = asset.get_batch_request({"year": "2019", "month": "01"})
+    return context, pandas_ds, asset, batch_request
+
+
+def sql_data(
+    context: AbstractDataContext,
+) -> tuple[AbstractDataContext, Datasource, DataAsset, BatchRequest]:
+    db_file = (
+        pathlib.Path(__file__)
+        / "../../../test_sets/taxi_yellow_tripdata_samples/sqlite/yellow_tripdata.db"
+    ).resolve()
+    datasource = context.sources.add_sqlite(
+        name="test_datasource",
+        connection_string=f"sqlite:///{db_file}",
+    )
+    asset = (
+        datasource.add_table_asset(
+            name="my_asset",
+            table_name="yellow_tripdata_sample_2019_01",
+        )
+        .add_year_and_month_splitter(column_name="pickup_datetime")
+        .add_sorters(["year", "month"])
+    )
+    batch_request = asset.get_batch_request({"year": 2019, "month": 1})
+    return context, datasource, asset, batch_request
+
+
+def spark_data(
+    context: AbstractDataContext,
+) -> tuple[AbstractDataContext, Datasource, DataAsset, BatchRequest]:
```

Comment:
Since we have this return type everywhere, maybe we use a namedtuple? Not super necessary but might be a nice quality-of-life improvement

Reply:
This was written in the same style as the analogous `pandas_data()` method.